pkg/envoy/lds/http_connection.go

```diff
@@ -0,0 +1,135 @@
+package lds
+
+import (
+	"fmt"
+
+	xds_route "github.com/envoyproxy/go-control-plane/envoy/config/route/v3"
+	xds_hcm "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/network/http_connection_manager/v3"
+	"github.com/envoyproxy/go-control-plane/pkg/wellknown"
+	"github.com/golang/protobuf/ptypes/wrappers"
+	"github.com/pkg/errors"
+
+	"github.com/openservicemesh/osm/pkg/auth"
+	"github.com/openservicemesh/osm/pkg/constants"
+	"github.com/openservicemesh/osm/pkg/envoy"
+)
+
+// connectionDirection defines, for filter terms, the direction of a connection from
+// the proxy's perspective while originating/terminating connections to/from other
+// proxies.
+type connectionDirection string
+
+const (
+	meshHTTPConnManagerStatPrefix       = "mesh-http-conn-manager"
+	prometheusHTTPConnManagerStatPrefix = "prometheus-http-conn-manager"
+	prometheusInboundVirtualHostName    = "prometheus-inbound-virtual-host"
+
+	// inbound defines in-mesh inbound or ingress traffic driections
+	inbound connectionDirection = "inbound"
+
+	// outbound defines in-mesh outbound or egress traffic directions
+	outbound connectionDirection = "outbound"
+)
+
+type httpConnManagerOptions struct {
+	direction         connectionDirection
+	rdsRoutConfigName string
+
+	// Additional filters
+	wasmStatsHeaders map[string]string
+	extAuthConfig    *auth.ExtAuthConfig
+
+	// Tracing options
+	enableTracing      bool
+	tracingAPIEndpoint string
+}
+
+func (options httpConnManagerOptions) buildFromOptions() (*xds_hcm.HttpConnectionManager, error) {
```

Comment:
`options` here is passed by value right, and therefore struct copy? wasmStatsHeader map and extAuthzConfig are already reference-copied if so I imagine, but not sure about connectionDirection

Reply:
`options` is passed by value to "indicate" the function does not mutate the input options. The pointer fields are used to represent optional fields in the struct, which is sort of a common pattern, for example with struct types that have json tags where using a pointer can differentiate between a value being unset vs being 0. `connectionDirection` is not a pointer, so it's copied.